// @uploadfly/react-widget/dist/react-widget.js
import z, { useState as X, useRef as Tr } from "react";
function Cr(e) {
  return e && e.__esModule && Object.prototype.hasOwnProperty.call(e, "default") ? e.default : e;
}
var Ie = { exports: {} }, ne = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var gt;
function Pr() {
  if (gt)
    return ne;
  gt = 1;
  var e = z, t = Symbol.for("react.element"), r = Symbol.for("react.fragment"), n = Object.prototype.hasOwnProperty, s = e.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, i = { key: !0, ref: !0, __self: !0, __source: !0 };
  function a(c, p, u) {
    var d, m = {}, R = null, g = null;
    u !== void 0 && (R = "" + u), p.key !== void 0 && (R = "" + p.key), p.ref !== void 0 && (g = p.ref);
    for (d in p)
      n.call(p, d) && !i.hasOwnProperty(d) && (m[d] = p[d]);
    if (c && c.defaultProps)
      for (d in p = c.defaultProps, p)
        m[d] === void 0 && (m[d] = p[d]);
    return { $$typeof: t, type: c, key: R, ref: g, props: m, _owner: s.current };
  }
  return ne.Fragment = r, ne.jsx = a, ne.jsxs = a, ne;
}
var oe = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var vt;
function Ar() {
  return vt || (vt = 1, process.env.NODE_ENV !== "production" && function() {
    var e = z, t = Symbol.for("react.element"), r = Symbol.for("react.portal"), n = Symbol.for("react.fragment"), s = Symbol.for("react.strict_mode"), i = Symbol.for("react.profiler"), a = Symbol.for("react.provider"), c = Symbol.for("react.context"), p = Symbol.for("react.forward_ref"), u = Symbol.for("react.suspense"), d = Symbol.for("react.suspense_list"), m = Symbol.for("react.memo"), R = Symbol.for("react.lazy"), g = Symbol.for("react.offscreen"), y = Symbol.iterator, b = "@@iterator";
    function D(o) {
      if (o === null || typeof o != "object")
        return null;
      var f = y && o[y] || o[b];
      return typeof f == "function" ? f : null;
    }
    var C = e.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function T(o) {
      {
        for (var f = arguments.length, h = new Array(f > 1 ? f - 1 : 0), E = 1; E < f; E++)
          h[E - 1] = arguments[E];
        $("error", o, h);
      }
    }
    function $(o, f, h) {
      {
        var E = C.ReactDebugCurrentFrame, S = E.getStackAddendum();
        S !== "" && (f += "%s", h = h.concat([S]));
        var x = h.map(function(w) {
          return String(w);
        });
        x.unshift("Warning: " + f), Function.prototype.apply.call(console[o], console, x);
      }
    }
    var Y = !1, xe = !1, ee = !1, N = !1, V = !1, Ke;
    Ke = Symbol.for("react.module.reference");
    function er(o) {
      return !!(typeof o == "string" || typeof o == "function" || o === n || o === i || V || o === s || o === u || o === d || N || o === g || Y || xe || ee || typeof o == "object" && o !== null && (o.$$typeof === R || o.$$typeof === m || o.$$typeof === a || o.$$typeof === c || o.$$typeof === p || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      o.$$typeof === Ke || o.getModuleId !== void 0));
    }
    function tr(o, f, h) {
      var E = o.displayName;
      if (E)
        return E;
      var S = f.displayName || f.name || "";
      return S !== "" ? h + "(" + S + ")" : h;
    }
    function Ge(o) {
      return o.displayName || "Context";
    }
    function I(o) {
      if (o == null)
        return null;
      if (typeof o.tag == "number" && T("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof o == "function")
        return o.displayName || o.name || null;
      if (typeof o == "string")
        return o;
      switch (o) {
        case n:
          return "Fragment";
        case r:
          return "Portal";
        case i:
          return "Profiler";
        case s:
          return "StrictMode";
        case u:
          return "Suspense";
        case d:
          return "SuspenseList";
      }
      if (typeof o == "object")
        switch (o.$$typeof) {
          case c:
            var f = o;
            return Ge(f) + ".Consumer";
          case a:
            var h = o;
            return Ge(h._context) + ".Provider";
          case p:
            return tr(o, o.render, "ForwardRef");
          case m:
            var E = o.displayName || null;
            return E !== null ? E : I(o.type) || "Memo";
          case R: {
            var S = o, x = S._payload, w = S._init;
            try {
              return I(w(x));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var J = Object.assign, te = 0, Xe, Ze, Qe, et, tt, rt, nt;
    function ot() {
    }
    ot.__reactDisabledLog = !0;
    function rr() {
      {
        if (te === 0) {
          Xe = console.log, Ze = console.info, Qe = console.warn, et = console.error, tt = console.group, rt = console.groupCollapsed, nt = console.groupEnd;
          var o = {
            configurable: !0,
            enumerable: !0,
            value: ot,
            writable: !0
          };
          Object.defineProperties(console, {
            info: o,
            log: o,
            warn: o,
            error: o,
            group: o,
            groupCollapsed: o,
            groupEnd: o
          });
        }
        te++;
      }
    }
    function nr() {
      {
        if (te--, te === 0) {
          var o = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: J({}, o, {
              value: Xe
            }),
            info: J({}, o, {
              value: Ze
            }),
            warn: J({}, o, {
              value: Qe
            }),
            error: J({}, o, {
              value: et
            }),
            group: J({}, o, {
              value: tt
            }),
            groupCollapsed: J({}, o, {
              value: rt
            }),
            groupEnd: J({}, o, {
              value: nt
            })
          });
        }
        te < 0 && T("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var Te = C.ReactCurrentDispatcher, Ce;
    function ce(o, f, h) {
      {
        if (Ce === void 0)
          try {
            throw Error();
          } catch (S) {
            var E = S.stack.trim().match(/\n( *(at )?)/);
            Ce = E && E[1] || "";
          }
        return `
` + Ce + o;
      }
    }
    var Pe = !1, fe;
    {
      var or = typeof WeakMap == "function" ? WeakMap : Map;
      fe = new or();
    }
    function st(o, f) {
      if (!o || Pe)
        return "";
      {
        var h = fe.get(o);
        if (h !== void 0)
          return h;
      }
      var E;
      Pe = !0;
      var S = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var x;
      x = Te.current, Te.current = null, rr();
      try {
        if (f) {
          var w = function() {
            throw Error();
          };
          if (Object.defineProperty(w.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(w, []);
            } catch (M) {
              E = M;
            }
            Reflect.construct(o, [], w);
          } else {
            try {
              w.call();
            } catch (M) {
              E = M;
            }
            o.call(w.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (M) {
            E = M;
          }
          o();
        }
      } catch (M) {
        if (M && E && typeof M.stack == "string") {
          for (var v = M.stack.split(`
`), F = E.stack.split(`
`), P = v.length - 1, A = F.length - 1; P >= 1 && A >= 0 && v[P] !== F[A]; )
            A--;
          for (; P >= 1 && A >= 0; P--, A--)
            if (v[P] !== F[A]) {
              if (P !== 1 || A !== 1)
                do
                  if (P--, A--, A < 0 || v[P] !== F[A]) {
                    var k = `
` + v[P].replace(" at new ", " at ");
                    return o.displayName && k.includes("<anonymous>") && (k = k.replace("<anonymous>", o.displayName)), typeof o == "function" && fe.set(o, k), k;
                  }
                while (P >= 1 && A >= 0);
              break;
            }
        }
      } finally {
        Pe = !1, Te.current = x, nr(), Error.prepareStackTrace = S;
      }
      var G = o ? o.displayName || o.name : "", bt = G ? ce(G) : "";
      return typeof o == "function" && fe.set(o, bt), bt;
    }
    function sr(o, f, h) {
      return st(o, !1);
    }
    function ir(o) {
      var f = o.prototype;
      return !!(f && f.isReactComponent);
    }
    function de(o, f, h) {
      if (o == null)
        return "";
      if (typeof o == "function")
        return st(o, ir(o));
      if (typeof o == "string")
        return ce(o);
      switch (o) {
        case u:
          return ce("Suspense");
        case d:
          return ce("SuspenseList");
      }
      if (typeof o == "object")
        switch (o.$$typeof) {
          case p:
            return sr(o.render);
          case m:
            return de(o.type, f, h);
          case R: {
            var E = o, S = E._payload, x = E._init;
            try {
              return de(x(S), f, h);
            } catch {
            }
          }
        }
      return "";
    }
    var pe = Object.prototype.hasOwnProperty, it = {}, at = C.ReactDebugCurrentFrame;
    function he(o) {
      if (o) {
        var f = o._owner, h = de(o.type, o._source, f ? f.type : null);
        at.setExtraStackFrame(h);
      } else
        at.setExtraStackFrame(null);
    }
    function ar(o, f, h, E, S) {
      {
        var x = Function.call.bind(pe);
        for (var w in o)
          if (x(o, w)) {
            var v = void 0;
            try {
              if (typeof o[w] != "function") {
                var F = Error((E || "React class") + ": " + h + " type `" + w + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof o[w] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw F.name = "Invariant Violation", F;
              }
              v = o[w](f, w, E, h, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (P) {
              v = P;
            }
            v && !(v instanceof Error) && (he(S), T("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", E || "React class", h, w, typeof v), he(null)), v instanceof Error && !(v.message in it) && (it[v.message] = !0, he(S), T("Failed %s type: %s", h, v.message), he(null));
          }
      }
    }
    var lr = Array.isArray;
    function Ae(o) {
      return lr(o);
    }
    function ur(o) {
      {
        var f = typeof Symbol == "function" && Symbol.toStringTag, h = f && o[Symbol.toStringTag] || o.constructor.name || "Object";
        return h;
      }
    }
    function cr(o) {
      try {
        return lt(o), !1;
      } catch {
        return !0;
      }
    }
    function lt(o) {
      return "" + o;
    }
    function ut(o) {
      if (cr(o))
        return T("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", ur(o)), lt(o);
    }
    var re = C.ReactCurrentOwner, fr = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, ct, ft, je;
    je = {};
    function dr(o) {
      if (pe.call(o, "ref")) {
        var f = Object.getOwnPropertyDescriptor(o, "ref").get;
        if (f && f.isReactWarning)
          return !1;
      }
      return o.ref !== void 0;
    }
    function pr(o) {
      if (pe.call(o, "key")) {
        var f = Object.getOwnPropertyDescriptor(o, "key").get;
        if (f && f.isReactWarning)
          return !1;
      }
      return o.key !== void 0;
    }
    function hr(o, f) {
      if (typeof o.ref == "string" && re.current && f && re.current.stateNode !== f) {
        var h = I(re.current.type);
        je[h] || (T('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', I(re.current.type), o.ref), je[h] = !0);
      }
    }
    function mr(o, f) {
      {
        var h = function() {
          ct || (ct = !0, T("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", f));
        };
        h.isReactWarning = !0, Object.defineProperty(o, "key", {
          get: h,
          configurable: !0
        });
      }
    }
    function yr(o, f) {
      {
        var h = function() {
          ft || (ft = !0, T("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", f));
        };
        h.isReactWarning = !0, Object.defineProperty(o, "ref", {
          get: h,
          configurable: !0
        });
      }
    }
    var Er = function(o, f, h, E, S, x, w) {
      var v = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: t,
        // Built-in properties that belong on the element
        type: o,
        key: f,
        ref: h,
        props: w,
        // Record the component responsible for creating this element.
        _owner: x
      };
      return v._store = {}, Object.defineProperty(v._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(v, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: E
      }), Object.defineProperty(v, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: S
      }), Object.freeze && (Object.freeze(v.props), Object.freeze(v)), v;
    };
    function br(o, f, h, E, S) {
      {
        var x, w = {}, v = null, F = null;
        h !== void 0 && (ut(h), v = "" + h), pr(f) && (ut(f.key), v = "" + f.key), dr(f) && (F = f.ref, hr(f, S));
        for (x in f)
          pe.call(f, x) && !fr.hasOwnProperty(x) && (w[x] = f[x]);
        if (o && o.defaultProps) {
          var P = o.defaultProps;
          for (x in P)
            w[x] === void 0 && (w[x] = P[x]);
        }
        if (v || F) {
          var A = typeof o == "function" ? o.displayName || o.name || "Unknown" : o;
          v && mr(w, A), F && yr(w, A);
        }
        return Er(o, v, F, S, E, re.current, w);
      }
    }
    var Ne = C.ReactCurrentOwner, dt = C.ReactDebugCurrentFrame;
    function K(o) {
      if (o) {
        var f = o._owner, h = de(o.type, o._source, f ? f.type : null);
        dt.setExtraStackFrame(h);
      } else
        dt.setExtraStackFrame(null);
    }
    var Fe;
    Fe = !1;
    function De(o) {
      return typeof o == "object" && o !== null && o.$$typeof === t;
    }
    function pt() {
      {
        if (Ne.current) {
          var o = I(Ne.current.type);
          if (o)
            return `

Check the render method of \`` + o + "`.";
        }
        return "";
      }
    }
    function gr(o) {
      {
        if (o !== void 0) {
          var f = o.fileName.replace(/^.*[\\\/]/, ""), h = o.lineNumber;
          return `

Check your code at ` + f + ":" + h + ".";
        }
        return "";
      }
    }
    var ht = {};
    function vr(o) {
      {
        var f = pt();
        if (!f) {
          var h = typeof o == "string" ? o : o.displayName || o.name;
          h && (f = `

Check the top-level render call using <` + h + ">.");
        }
        return f;
      }
    }
    function mt(o, f) {
      {
        if (!o._store || o._store.validated || o.key != null)
          return;
        o._store.validated = !0;
        var h = vr(f);
        if (ht[h])
          return;
        ht[h] = !0;
        var E = "";
        o && o._owner && o._owner !== Ne.current && (E = " It was passed a child from " + I(o._owner.type) + "."), K(o), T('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', h, E), K(null);
      }
    }
    function yt(o, f) {
      {
        if (typeof o != "object")
          return;
        if (Ae(o))
          for (var h = 0; h < o.length; h++) {
            var E = o[h];
            De(E) && mt(E, f);
          }
        else if (De(o))
          o._store && (o._store.validated = !0);
        else if (o) {
          var S = D(o);
          if (typeof S == "function" && S !== o.entries)
            for (var x = S.call(o), w; !(w = x.next()).done; )
              De(w.value) && mt(w.value, f);
        }
      }
    }
    function wr(o) {
      {
        var f = o.type;
        if (f == null || typeof f == "string")
          return;
        var h;
        if (typeof f == "function")
          h = f.propTypes;
        else if (typeof f == "object" && (f.$$typeof === p || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        f.$$typeof === m))
          h = f.propTypes;
        else
          return;
        if (h) {
          var E = I(f);
          ar(h, o.props, "prop", E, o);
        } else if (f.PropTypes !== void 0 && !Fe) {
          Fe = !0;
          var S = I(f);
          T("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", S || "Unknown");
        }
        typeof f.getDefaultProps == "function" && !f.getDefaultProps.isReactClassApproved && T("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function Rr(o) {
      {
        for (var f = Object.keys(o.props), h = 0; h < f.length; h++) {
          var E = f[h];
          if (E !== "children" && E !== "key") {
            K(o), T("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", E), K(null);
            break;
          }
        }
        o.ref !== null && (K(o), T("Invalid attribute `ref` supplied to `React.Fragment`."), K(null));
      }
    }
    function Et(o, f, h, E, S, x) {
      {
        var w = er(o);
        if (!w) {
          var v = "";
          (o === void 0 || typeof o == "object" && o !== null && Object.keys(o).length === 0) && (v += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var F = gr(S);
          F ? v += F : v += pt();
          var P;
          o === null ? P = "null" : Ae(o) ? P = "array" : o !== void 0 && o.$$typeof === t ? (P = "<" + (I(o.type) || "Unknown") + " />", v = " Did you accidentally export a JSX literal instead of a component?") : P = typeof o, T("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", P, v);
        }
        var A = br(o, f, h, S, x);
        if (A == null)
          return A;
        if (w) {
          var k = f.children;
          if (k !== void 0)
            if (E)
              if (Ae(k)) {
                for (var G = 0; G < k.length; G++)
                  yt(k[G], o);
                Object.freeze && Object.freeze(k);
              } else
                T("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              yt(k, o);
        }
        return o === n ? Rr(A) : wr(A), A;
      }
    }
    function _r(o, f, h) {
      return Et(o, f, h, !0);
    }
    function Or(o, f, h) {
      return Et(o, f, h, !1);
    }
    var Sr = Or, xr = _r;
    oe.Fragment = n, oe.jsx = Sr, oe.jsxs = xr;
  }()), oe;
}
process.env.NODE_ENV === "production" ? Ie.exports = Pr() : Ie.exports = Ar();
var O = Ie.exports, Nt = {
  color: void 0,
  size: void 0,
  className: void 0,
  style: void 0,
  attr: void 0
}, wt = z.createContext && z.createContext(Nt), W = globalThis && globalThis.__assign || function() {
  return W = Object.assign || function(e) {
    for (var t, r = 1, n = arguments.length; r < n; r++) {
      t = arguments[r];
      for (var s in t)
        Object.prototype.hasOwnProperty.call(t, s) && (e[s] = t[s]);
    }
    return e;
  }, W.apply(this, arguments);
}, jr = globalThis && globalThis.__rest || function(e, t) {
  var r = {};
  for (var n in e)
    Object.prototype.hasOwnProperty.call(e, n) && t.indexOf(n) < 0 && (r[n] = e[n]);
  if (e != null && typeof Object.getOwnPropertySymbols == "function")
    for (var s = 0, n = Object.getOwnPropertySymbols(e); s < n.length; s++)
      t.indexOf(n[s]) < 0 && Object.prototype.propertyIsEnumerable.call(e, n[s]) && (r[n[s]] = e[n[s]]);
  return r;
};
function Ft(e) {
  return e && e.map(function(t, r) {
    return z.createElement(t.tag, W({
      key: r
    }, t.attr), Ft(t.child));
  });
}
function ae(e) {
  return function(t) {
    return z.createElement(Nr, W({
      attr: W({}, e.attr)
    }, t), Ft(e.child));
  };
}
function Nr(e) {
  var t = function(r) {
    var n = e.attr, s = e.size, i = e.title, a = jr(e, ["attr", "size", "title"]), c = s || r.size || "1em", p;
    return r.className && (p = r.className), e.className && (p = (p ? p + " " : "") + e.className), z.createElement("svg", W({
      stroke: "currentColor",
      fill: "currentColor",
      strokeWidth: "0"
    }, r.attr, n, a, {
      className: p,
      style: W(W({
        color: e.color || r.color
      }, r.style), e.style),
      height: c,
      width: c,
      xmlns: "http://www.w3.org/2000/svg"
    }), i && z.createElement("title", null, i), e.children);
  };
  return wt !== void 0 ? z.createElement(wt.Consumer, null, function(r) {
    return t(r);
  }) : t(Nt);
}
function Fr(e) {
  return ae({ tag: "svg", attr: { viewBox: "0 0 24 24" }, child: [{ tag: "path", attr: { fill: "none", d: "M0 0h24v24H0z" } }, { tag: "path", attr: { d: "M5 20h14v-2H5v2zm0-10h4v6h6v-6h4l-7-7-7 7z" } }] })(e);
}
function Dr(e) {
  return ae({ tag: "svg", attr: { viewBox: "0 0 24 24" }, child: [{ tag: "path", attr: { fill: "none", d: "M0 0h24v24H0z" } }, { tag: "path", attr: { d: "M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z" } }] })(e);
}
var Dt = { exports: {} };
(function(e) {
  (function(t) {
    if (e.exports)
      return e.exports = t();
    this.filesize = t();
  })(function() {
    var t = "BKMGTPEZY".split("");
    function r(n, s) {
      return n && n.toLowerCase() === s.toLowerCase();
    }
    return function(s, i) {
      return s = typeof s == "number" ? s : 0, i = i || {}, i.fixed = typeof i.fixed == "number" ? i.fixed : 2, i.spacer = typeof i.spacer == "string" ? i.spacer : " ", i.calculate = function(a) {
        var c = r(a, "si") ? ["k", "B"] : ["K", "iB"], p = r(a, "si") ? 1e3 : 1024, u = Math.log(s) / Math.log(p) | 0, d = s / Math.pow(p, u), m = d.toFixed(i.fixed), R;
        return u - 1 < 3 && !r(a, "si") && r(a, "jedec") && (c[1] = "B"), R = u ? (c[0] + "MGTPEZY")[u - 1] + c[1] : (m | 0) === 1 ? "Byte" : "Bytes", {
          suffix: R,
          magnitude: u,
          result: d,
          fixed: m,
          bits: { result: d / 8, fixed: (d / 8).toFixed(i.fixed) }
        };
      }, i.to = function(a, c) {
        var p = r(c, "si") ? 1e3 : 1024, u = t.indexOf(typeof a == "string" ? a[0].toUpperCase() : "B"), d = s;
        if (u === -1 || u === 0)
          return d.toFixed(2);
        for (; u > 0; u--)
          d /= p;
        return d.toFixed(2);
      }, i.human = function(a) {
        var c = i.calculate(a);
        return c.fixed + i.spacer + c.suffix;
      }, i;
    };
  });
})(Dt);
var kr = Dt.exports;
const Br = /* @__PURE__ */ Cr(kr);
function kt(e, t) {
  return function() {
    return e.apply(t, arguments);
  };
}
const { toString: Lr } = Object.prototype, { getPrototypeOf: qe } = Object, ve = ((e) => (t) => {
  const r = Lr.call(t);
  return e[r] || (e[r] = r.slice(8, -1).toLowerCase());
})(/* @__PURE__ */ Object.create(null)), U = (e) => (e = e.toLowerCase(), (t) => ve(t) === e), we = (e) => (t) => typeof t === e, { isArray: Q } = Array, ie = we("undefined");
function Ur(e) {
  return e !== null && !ie(e) && e.constructor !== null && !ie(e.constructor) && B(e.constructor.isBuffer) && e.constructor.isBuffer(e);
}
const Bt = U("ArrayBuffer");
function Ir(e) {
  let t;
  return typeof ArrayBuffer < "u" && ArrayBuffer.isView ? t = ArrayBuffer.isView(e) : t = e && e.buffer && Bt(e.buffer), t;
}
const Mr = we("string"), B = we("function"), Lt = we("number"), Re = (e) => e !== null && typeof e == "object", zr = (e) => e === !0 || e === !1, me = (e) => {
  if (ve(e) !== "object")
    return !1;
  const t = qe(e);
  return (t === null || t === Object.prototype || Object.getPrototypeOf(t) === null) && !(Symbol.toStringTag in e) && !(Symbol.iterator in e);
}, Hr = U("Date"), $r = U("File"), qr = U("Blob"), Wr = U("FileList"), Vr = (e) => Re(e) && B(e.pipe), Jr = (e) => {
  let t;
  return e && (typeof FormData == "function" && e instanceof FormData || B(e.append) && ((t = ve(e)) === "formdata" || // detect form-data instance
  t === "object" && B(e.toString) && e.toString() === "[object FormData]"));
}, Yr = U("URLSearchParams"), Kr = (e) => e.trim ? e.trim() : e.replace(/^[\s\uFEFF\xA0]+|[\s\uFEFF\xA0]+$/g, "");
function le(e, t, { allOwnKeys: r = !1 } = {}) {
  if (e === null || typeof e > "u")
    return;
  let n, s;
  if (typeof e != "object" && (e = [e]), Q(e))
    for (n = 0, s = e.length; n < s; n++)
      t.call(null, e[n], n, e);
  else {
    const i = r ? Object.getOwnPropertyNames(e) : Object.keys(e), a = i.length;
    let c;
    for (n = 0; n < a; n++)
      c = i[n], t.call(null, e[c], c, e);
  }
}
function Ut(e, t) {
  t = t.toLowerCase();
  const r = Object.keys(e);
  let n = r.length, s;
  for (; n-- > 0; )
    if (s = r[n], t === s.toLowerCase())
      return s;
  return null;
}
const It = (() => typeof globalThis < "u" ? globalThis : typeof self < "u" ? self : typeof window < "u" ? window : global)(), Mt = (e) => !ie(e) && e !== It;
function Me() {
  const { caseless: e } = Mt(this) && this || {}, t = {}, r = (n, s) => {
    const i = e && Ut(t, s) || s;
    me(t[i]) && me(n) ? t[i] = Me(t[i], n) : me(n) ? t[i] = Me({}, n) : Q(n) ? t[i] = n.slice() : t[i] = n;
  };
  for (let n = 0, s = arguments.length; n < s; n++)
    arguments[n] && le(arguments[n], r);
  return t;
}
const Gr = (e, t, r, { allOwnKeys: n } = {}) => (le(t, (s, i) => {
  r && B(s) ? e[i] = kt(s, r) : e[i] = s;
}, { allOwnKeys: n }), e), Xr = (e) => (e.charCodeAt(0) === 65279 && (e = e.slice(1)), e), Zr = (e, t, r, n) => {
  e.prototype = Object.create(t.prototype, n), e.prototype.constructor = e, Object.defineProperty(e, "super", {
    value: t.prototype
  }), r && Object.assign(e.prototype, r);
}, Qr = (e, t, r, n) => {
  let s, i, a;
  const c = {};
  if (t = t || {}, e == null)
    return t;
  do {
    for (s = Object.getOwnPropertyNames(e), i = s.length; i-- > 0; )
      a = s[i], (!n || n(a, e, t)) && !c[a] && (t[a] = e[a], c[a] = !0);
    e = r !== !1 && qe(e);
  } while (e && (!r || r(e, t)) && e !== Object.prototype);
  return t;
}, en = (e, t, r) => {
  e = String(e), (r === void 0 || r > e.length) && (r = e.length), r -= t.length;
  const n = e.indexOf(t, r);
  return n !== -1 && n === r;
}, tn = (e) => {
  if (!e)
    return null;
  if (Q(e))
    return e;
  let t = e.length;
  if (!Lt(t))
    return null;
  const r = new Array(t);
  for (; t-- > 0; )
    r[t] = e[t];
  return r;
}, rn = ((e) => (t) => e && t instanceof e)(typeof Uint8Array < "u" && qe(Uint8Array)), nn = (e, t) => {
  const n = (e && e[Symbol.iterator]).call(e);
  let s;
  for (; (s = n.next()) && !s.done; ) {
    const i = s.value;
    t.call(e, i[0], i[1]);
  }
}, on = (e, t) => {
  let r;
  const n = [];
  for (; (r = e.exec(t)) !== null; )
    n.push(r);
  return n;
}, sn = U("HTMLFormElement"), an = (e) => e.toLowerCase().replace(
  /[-_\s]([a-z\d])(\w*)/g,
  function(r, n, s) {
    return n.toUpperCase() + s;
  }
), Rt = (({ hasOwnProperty: e }) => (t, r) => e.call(t, r))(Object.prototype), ln = U("RegExp"), zt = (e, t) => {
  const r = Object.getOwnPropertyDescriptors(e), n = {};
  le(r, (s, i) => {
    t(s, i, e) !== !1 && (n[i] = s);
  }), Object.defineProperties(e, n);
}, un = (e) => {
  zt(e, (t, r) => {
    if (B(e) && ["arguments", "caller", "callee"].indexOf(r) !== -1)
      return !1;
    const n = e[r];
    if (B(n)) {
      if (t.enumerable = !1, "writable" in t) {
        t.writable = !1;
        return;
      }
      t.set || (t.set = () => {
        throw Error("Can not rewrite read-only method '" + r + "'");
      });
    }
  });
}, cn = (e, t) => {
  const r = {}, n = (s) => {
    s.forEach((i) => {
      r[i] = !0;
    });
  };
  return Q(e) ? n(e) : n(String(e).split(t)), r;
}, fn = () => {
}, dn = (e, t) => (e = +e, Number.isFinite(e) ? e : t), ke = "abcdefghijklmnopqrstuvwxyz", _t = "0123456789", Ht = {
  DIGIT: _t,
  ALPHA: ke,
  ALPHA_DIGIT: ke + ke.toUpperCase() + _t
}, pn = (e = 16, t = Ht.ALPHA_DIGIT) => {
  let r = "";
  const { length: n } = t;
  for (; e--; )
    r += t[Math.random() * n | 0];
  return r;
};
function hn(e) {
  return !!(e && B(e.append) && e[Symbol.toStringTag] === "FormData" && e[Symbol.iterator]);
}
const mn = (e) => {
  const t = new Array(10), r = (n, s) => {
    if (Re(n)) {
      if (t.indexOf(n) >= 0)
        return;
      if (!("toJSON" in n)) {
        t[s] = n;
        const i = Q(n) ? [] : {};
        return le(n, (a, c) => {
          const p = r(a, s + 1);
          !ie(p) && (i[c] = p);
        }), t[s] = void 0, i;
      }
    }
    return n;
  };
  return r(e, 0);
}, yn = U("AsyncFunction"), En = (e) => e && (Re(e) || B(e)) && B(e.then) && B(e.catch), l = {
  isArray: Q,
  isArrayBuffer: Bt,
  isBuffer: Ur,
  isFormData: Jr,
  isArrayBufferView: Ir,
  isString: Mr,
  isNumber: Lt,
  isBoolean: zr,
  isObject: Re,
  isPlainObject: me,
  isUndefined: ie,
  isDate: Hr,
  isFile: $r,
  isBlob: qr,
  isRegExp: ln,
  isFunction: B,
  isStream: Vr,
  isURLSearchParams: Yr,
  isTypedArray: rn,
  isFileList: Wr,
  forEach: le,
  merge: Me,
  extend: Gr,
  trim: Kr,
  stripBOM: Xr,
  inherits: Zr,
  toFlatObject: Qr,
  kindOf: ve,
  kindOfTest: U,
  endsWith: en,
  toArray: tn,
  forEachEntry: nn,
  matchAll: on,
  isHTMLForm: sn,
  hasOwnProperty: Rt,
  hasOwnProp: Rt,
  // an alias to avoid ESLint no-prototype-builtins detection
  reduceDescriptors: zt,
  freezeMethods: un,
  toObjectSet: cn,
  toCamelCase: an,
  noop: fn,
  toFiniteNumber: dn,
  findKey: Ut,
  global: It,
  isContextDefined: Mt,
  ALPHABET: Ht,
  generateString: pn,
  isSpecCompliantForm: hn,
  toJSONObject: mn,
  isAsyncFn: yn,
  isThenable: En
};
function _(e, t, r, n, s) {
  Error.call(this), Error.captureStackTrace ? Error.captureStackTrace(this, this.constructor) : this.stack = new Error().stack, this.message = e, this.name = "AxiosError", t && (this.code = t), r && (this.config = r), n && (this.request = n), s && (this.response = s);
}
l.inherits(_, Error, {
  toJSON: function() {
    return {
      // Standard
      message: this.message,
      name: this.name,
      // Microsoft
      description: this.description,
      number: this.number,
      // Mozilla
      fileName: this.fileName,
      lineNumber: this.lineNumber,
      columnNumber: this.columnNumber,
      stack: this.stack,
      // Axios
      config: l.toJSONObject(this.config),
      code: this.code,
      status: this.response && this.response.status ? this.response.status : null
    };
  }
});
const $t = _.prototype, qt = {};
[
  "ERR_BAD_OPTION_VALUE",
  "ERR_BAD_OPTION",
  "ECONNABORTED",
  "ETIMEDOUT",
  "ERR_NETWORK",
  "ERR_FR_TOO_MANY_REDIRECTS",
  "ERR_DEPRECATED",
  "ERR_BAD_RESPONSE",
  "ERR_BAD_REQUEST",
  "ERR_CANCELED",
  "ERR_NOT_SUPPORT",
  "ERR_INVALID_URL"
  // eslint-disable-next-line func-names
].forEach((e) => {
  qt[e] = { value: e };
});
Object.defineProperties(_, qt);
Object.defineProperty($t, "isAxiosError", { value: !0 });
_.from = (e, t, r, n, s, i) => {
  const a = Object.create($t);
  return l.toFlatObject(e, a, function(p) {
    return p !== Error.prototype;
  }, (c) => c !== "isAxiosError"), _.call(a, e.message, t, r, n, s), a.cause = e, a.name = e.name, i && Object.assign(a, i), a;
};
const bn = null;
function ze(e) {
  return l.isPlainObject(e) || l.isArray(e);
}
function Wt(e) {
  return l.endsWith(e, "[]") ? e.slice(0, -2) : e;
}
function Ot(e, t, r) {
  return e ? e.concat(t).map(function(s, i) {
    return s = Wt(s), !r && i ? "[" + s + "]" : s;
  }).join(r ? "." : "") : t;
}
function gn(e) {
  return l.isArray(e) && !e.some(ze);
}
const vn = l.toFlatObject(l, {}, null, function(t) {
  return /^is[A-Z]/.test(t);
});
function _e(e, t, r) {
  if (!l.isObject(e))
    throw new TypeError("target must be an object");
  t = t || new FormData(), r = l.toFlatObject(r, {
    metaTokens: !0,
    dots: !1,
    indexes: !1
  }, !1, function(b, D) {
    return !l.isUndefined(D[b]);
  });
  const n = r.metaTokens, s = r.visitor || d, i = r.dots, a = r.indexes, p = (r.Blob || typeof Blob < "u" && Blob) && l.isSpecCompliantForm(t);
  if (!l.isFunction(s))
    throw new TypeError("visitor must be a function");
  function u(y) {
    if (y === null)
      return "";
    if (l.isDate(y))
      return y.toISOString();
    if (!p && l.isBlob(y))
      throw new _("Blob is not supported. Use a Buffer instead.");
    return l.isArrayBuffer(y) || l.isTypedArray(y) ? p && typeof Blob == "function" ? new Blob([y]) : Buffer.from(y) : y;
  }
  function d(y, b, D) {
    let C = y;
    if (y && !D && typeof y == "object") {
      if (l.endsWith(b, "{}"))
        b = n ? b : b.slice(0, -2), y = JSON.stringify(y);
      else if (l.isArray(y) && gn(y) || (l.isFileList(y) || l.endsWith(b, "[]")) && (C = l.toArray(y)))
        return b = Wt(b), C.forEach(function($, Y) {
          !(l.isUndefined($) || $ === null) && t.append(
            // eslint-disable-next-line no-nested-ternary
            a === !0 ? Ot([b], Y, i) : a === null ? b : b + "[]",
            u($)
          );
        }), !1;
    }
    return ze(y) ? !0 : (t.append(Ot(D, b, i), u(y)), !1);
  }
  const m = [], R = Object.assign(vn, {
    defaultVisitor: d,
    convertValue: u,
    isVisitable: ze
  });
  function g(y, b) {
    if (!l.isUndefined(y)) {
      if (m.indexOf(y) !== -1)
        throw Error("Circular reference detected in " + b.join("."));
      m.push(y), l.forEach(y, function(C, T) {
        (!(l.isUndefined(C) || C === null) && s.call(
          t,
          C,
          l.isString(T) ? T.trim() : T,
          b,
          R
        )) === !0 && g(C, b ? b.concat(T) : [T]);
      }), m.pop();
    }
  }
  if (!l.isObject(e))
    throw new TypeError("data must be an object");
  return g(e), t;
}
function St(e) {
  const t = {
    "!": "%21",
    "'": "%27",
    "(": "%28",
    ")": "%29",
    "~": "%7E",
    "%20": "+",
    "%00": "\0"
  };
  return encodeURIComponent(e).replace(/[!'()~]|%20|%00/g, function(n) {
    return t[n];
  });
}
function We(e, t) {
  this._pairs = [], e && _e(e, this, t);
}
const Vt = We.prototype;
Vt.append = function(t, r) {
  this._pairs.push([t, r]);
};
Vt.toString = function(t) {
  const r = t ? function(n) {
    return t.call(this, n, St);
  } : St;
  return this._pairs.map(function(s) {
    return r(s[0]) + "=" + r(s[1]);
  }, "").join("&");
};
function wn(e) {
  return encodeURIComponent(e).replace(/%3A/gi, ":").replace(/%24/g, "$").replace(/%2C/gi, ",").replace(/%20/g, "+").replace(/%5B/gi, "[").replace(/%5D/gi, "]");
}
function Jt(e, t, r) {
  if (!t)
    return e;
  const n = r && r.encode || wn, s = r && r.serialize;
  let i;
  if (s ? i = s(t, r) : i = l.isURLSearchParams(t) ? t.toString() : new We(t, r).toString(n), i) {
    const a = e.indexOf("#");
    a !== -1 && (e = e.slice(0, a)), e += (e.indexOf("?") === -1 ? "?" : "&") + i;
  }
  return e;
}
class Rn {
  constructor() {
    this.handlers = [];
  }
  /**
   * Add a new interceptor to the stack
   *
   * @param {Function} fulfilled The function to handle `then` for a `Promise`
   * @param {Function} rejected The function to handle `reject` for a `Promise`
   *
   * @return {Number} An ID used to remove interceptor later
   */
  use(t, r, n) {
    return this.handlers.push({
      fulfilled: t,
      rejected: r,
      synchronous: n ? n.synchronous : !1,
      runWhen: n ? n.runWhen : null
    }), this.handlers.length - 1;
  }
  /**
   * Remove an interceptor from the stack
   *
   * @param {Number} id The ID that was returned by `use`
   *
   * @returns {Boolean} `true` if the interceptor was removed, `false` otherwise
   */
  eject(t) {
    this.handlers[t] && (this.handlers[t] = null);
  }
  /**
   * Clear all interceptors from the stack
   *
   * @returns {void}
   */
  clear() {
    this.handlers && (this.handlers = []);
  }
  /**
   * Iterate over all the registered interceptors
   *
   * This method is particularly useful for skipping over any
   * interceptors that may have become `null` calling `eject`.
   *
   * @param {Function} fn The function to call for each interceptor
   *
   * @returns {void}
   */
  forEach(t) {
    l.forEach(this.handlers, function(n) {
      n !== null && t(n);
    });
  }
}
const xt = Rn, Yt = {
  silentJSONParsing: !0,
  forcedJSONParsing: !0,
  clarifyTimeoutError: !1
}, _n = typeof URLSearchParams < "u" ? URLSearchParams : We, On = typeof FormData < "u" ? FormData : null, Sn = typeof Blob < "u" ? Blob : null, xn = (() => {
  let e;
  return typeof navigator < "u" && ((e = navigator.product) === "ReactNative" || e === "NativeScript" || e === "NS") ? !1 : typeof window < "u" && typeof document < "u";
})(), Tn = (() => typeof WorkerGlobalScope < "u" && // eslint-disable-next-line no-undef
self instanceof WorkerGlobalScope && typeof self.importScripts == "function")(), L = {
  isBrowser: !0,
  classes: {
    URLSearchParams: _n,
    FormData: On,
    Blob: Sn
  },
  isStandardBrowserEnv: xn,
  isStandardBrowserWebWorkerEnv: Tn,
  protocols: ["http", "https", "file", "blob", "url", "data"]
};
function Cn(e, t) {
  return _e(e, new L.classes.URLSearchParams(), Object.assign({
    visitor: function(r, n, s, i) {
      return L.isNode && l.isBuffer(r) ? (this.append(n, r.toString("base64")), !1) : i.defaultVisitor.apply(this, arguments);
    }
  }, t));
}
function Pn(e) {
  return l.matchAll(/\w+|\[(\w*)]/g, e).map((t) => t[0] === "[]" ? "" : t[1] || t[0]);
}
function An(e) {
  const t = {}, r = Object.keys(e);
  let n;
  const s = r.length;
  let i;
  for (n = 0; n < s; n++)
    i = r[n], t[i] = e[i];
  return t;
}
function Kt(e) {
  function t(r, n, s, i) {
    let a = r[i++];
    const c = Number.isFinite(+a), p = i >= r.length;
    return a = !a && l.isArray(s) ? s.length : a, p ? (l.hasOwnProp(s, a) ? s[a] = [s[a], n] : s[a] = n, !c) : ((!s[a] || !l.isObject(s[a])) && (s[a] = []), t(r, n, s[a], i) && l.isArray(s[a]) && (s[a] = An(s[a])), !c);
  }
  if (l.isFormData(e) && l.isFunction(e.entries)) {
    const r = {};
    return l.forEachEntry(e, (n, s) => {
      t(Pn(n), s, r, 0);
    }), r;
  }
  return null;
}
const jn = {
  "Content-Type": void 0
};
function Nn(e, t, r) {
  if (l.isString(e))
    try {
      return (t || JSON.parse)(e), l.trim(e);
    } catch (n) {
      if (n.name !== "SyntaxError")
        throw n;
    }
  return (r || JSON.stringify)(e);
}
const Oe = {
  transitional: Yt,
  adapter: ["xhr", "http"],
  transformRequest: [function(t, r) {
    const n = r.getContentType() || "", s = n.indexOf("application/json") > -1, i = l.isObject(t);
    if (i && l.isHTMLForm(t) && (t = new FormData(t)), l.isFormData(t))
      return s && s ? JSON.stringify(Kt(t)) : t;
    if (l.isArrayBuffer(t) || l.isBuffer(t) || l.isStream(t) || l.isFile(t) || l.isBlob(t))
      return t;
    if (l.isArrayBufferView(t))
      return t.buffer;
    if (l.isURLSearchParams(t))
      return r.setContentType("application/x-www-form-urlencoded;charset=utf-8", !1), t.toString();
    let c;
    if (i) {
      if (n.indexOf("application/x-www-form-urlencoded") > -1)
        return Cn(t, this.formSerializer).toString();
      if ((c = l.isFileList(t)) || n.indexOf("multipart/form-data") > -1) {
        const p = this.env && this.env.FormData;
        return _e(
          c ? { "files[]": t } : t,
          p && new p(),
          this.formSerializer
        );
      }
    }
    return i || s ? (r.setContentType("application/json", !1), Nn(t)) : t;
  }],
  transformResponse: [function(t) {
    const r = this.transitional || Oe.transitional, n = r && r.forcedJSONParsing, s = this.responseType === "json";
    if (t && l.isString(t) && (n && !this.responseType || s)) {
      const a = !(r && r.silentJSONParsing) && s;
      try {
        return JSON.parse(t);
      } catch (c) {
        if (a)
          throw c.name === "SyntaxError" ? _.from(c, _.ERR_BAD_RESPONSE, this, null, this.response) : c;
      }
    }
    return t;
  }],
  /**
   * A timeout in milliseconds to abort a request. If set to 0 (default) a
   * timeout is not created.
   */
  timeout: 0,
  xsrfCookieName: "XSRF-TOKEN",
  xsrfHeaderName: "X-XSRF-TOKEN",
  maxContentLength: -1,
  maxBodyLength: -1,
  env: {
    FormData: L.classes.FormData,
    Blob: L.classes.Blob
  },
  validateStatus: function(t) {
    return t >= 200 && t < 300;
  },
  headers: {
    common: {
      Accept: "application/json, text/plain, */*"
    }
  }
};
l.forEach(["delete", "get", "head"], function(t) {
  Oe.headers[t] = {};
});
l.forEach(["post", "put", "patch"], function(t) {
  Oe.headers[t] = l.merge(jn);
});
const Ve = Oe, Fn = l.toObjectSet([
  "age",
  "authorization",
  "content-length",
  "content-type",
  "etag",
  "expires",
  "from",
  "host",
  "if-modified-since",
  "if-unmodified-since",
  "last-modified",
  "location",
  "max-forwards",
  "proxy-authorization",
  "referer",
  "retry-after",
  "user-agent"
]), Dn = (e) => {
  const t = {};
  let r, n, s;
  return e && e.split(`
`).forEach(function(a) {
    s = a.indexOf(":"), r = a.substring(0, s).trim().toLowerCase(), n = a.substring(s + 1).trim(), !(!r || t[r] && Fn[r]) && (r === "set-cookie" ? t[r] ? t[r].push(n) : t[r] = [n] : t[r] = t[r] ? t[r] + ", " + n : n);
  }), t;
}, Tt = Symbol("internals");
function se(e) {
  return e && String(e).trim().toLowerCase();
}
function ye(e) {
  return e === !1 || e == null ? e : l.isArray(e) ? e.map(ye) : String(e);
}
function kn(e) {
  const t = /* @__PURE__ */ Object.create(null), r = /([^\s,;=]+)\s*(?:=\s*([^,;]+))?/g;
  let n;
  for (; n = r.exec(e); )
    t[n[1]] = n[2];
  return t;
}
const Bn = (e) => /^[-_a-zA-Z0-9^`|~,!#$%&'*+.]+$/.test(e.trim());
function Be(e, t, r, n, s) {
  if (l.isFunction(n))
    return n.call(this, t, r);
  if (s && (t = r), !!l.isString(t)) {
    if (l.isString(n))
      return t.indexOf(n) !== -1;
    if (l.isRegExp(n))
      return n.test(t);
  }
}
function Ln(e) {
  return e.trim().toLowerCase().replace(/([a-z\d])(\w*)/g, (t, r, n) => r.toUpperCase() + n);
}
function Un(e, t) {
  const r = l.toCamelCase(" " + t);
  ["get", "set", "has"].forEach((n) => {
    Object.defineProperty(e, n + r, {
      value: function(s, i, a) {
        return this[n].call(this, t, s, i, a);
      },
      configurable: !0
    });
  });
}
class Se {
  constructor(t) {
    t && this.set(t);
  }
  set(t, r, n) {
    const s = this;
    function i(c, p, u) {
      const d = se(p);
      if (!d)
        throw new Error("header name must be a non-empty string");
      const m = l.findKey(s, d);
      (!m || s[m] === void 0 || u === !0 || u === void 0 && s[m] !== !1) && (s[m || p] = ye(c));
    }
    const a = (c, p) => l.forEach(c, (u, d) => i(u, d, p));
    return l.isPlainObject(t) || t instanceof this.constructor ? a(t, r) : l.isString(t) && (t = t.trim()) && !Bn(t) ? a(Dn(t), r) : t != null && i(r, t, n), this;
  }
  get(t, r) {
    if (t = se(t), t) {
      const n = l.findKey(this, t);
      if (n) {
        const s = this[n];
        if (!r)
          return s;
        if (r === !0)
          return kn(s);
        if (l.isFunction(r))
          return r.call(this, s, n);
        if (l.isRegExp(r))
          return r.exec(s);
        throw new TypeError("parser must be boolean|regexp|function");
      }
    }
  }
  has(t, r) {
    if (t = se(t), t) {
      const n = l.findKey(this, t);
      return !!(n && this[n] !== void 0 && (!r || Be(this, this[n], n, r)));
    }
    return !1;
  }
  delete(t, r) {
    const n = this;
    let s = !1;
    function i(a) {
      if (a = se(a), a) {
        const c = l.findKey(n, a);
        c && (!r || Be(n, n[c], c, r)) && (delete n[c], s = !0);
      }
    }
    return l.isArray(t) ? t.forEach(i) : i(t), s;
  }
  clear(t) {
    const r = Object.keys(this);
    let n = r.length, s = !1;
    for (; n--; ) {
      const i = r[n];
      (!t || Be(this, this[i], i, t, !0)) && (delete this[i], s = !0);
    }
    return s;
  }
  normalize(t) {
    const r = this, n = {};
    return l.forEach(this, (s, i) => {
      const a = l.findKey(n, i);
      if (a) {
        r[a] = ye(s), delete r[i];
        return;
      }
      const c = t ? Ln(i) : String(i).trim();
      c !== i && delete r[i], r[c] = ye(s), n[c] = !0;
    }), this;
  }
  concat(...t) {
    return this.constructor.concat(this, ...t);
  }
  toJSON(t) {
    const r = /* @__PURE__ */ Object.create(null);
    return l.forEach(this, (n, s) => {
      n != null && n !== !1 && (r[s] = t && l.isArray(n) ? n.join(", ") : n);
    }), r;
  }
  [Symbol.iterator]() {
    return Object.entries(this.toJSON())[Symbol.iterator]();
  }
  toString() {
    return Object.entries(this.toJSON()).map(([t, r]) => t + ": " + r).join(`
`);
  }
  get [Symbol.toStringTag]() {
    return "AxiosHeaders";
  }
  static from(t) {
    return t instanceof this ? t : new this(t);
  }
  static concat(t, ...r) {
    const n = new this(t);
    return r.forEach((s) => n.set(s)), n;
  }
  static accessor(t) {
    const n = (this[Tt] = this[Tt] = {
      accessors: {}
    }).accessors, s = this.prototype;
    function i(a) {
      const c = se(a);
      n[c] || (Un(s, a), n[c] = !0);
    }
    return l.isArray(t) ? t.forEach(i) : i(t), this;
  }
}
Se.accessor(["Content-Type", "Content-Length", "Accept", "Accept-Encoding", "User-Agent", "Authorization"]);
l.freezeMethods(Se.prototype);
l.freezeMethods(Se);
const H = Se;
function Le(e, t) {
  const r = this || Ve, n = t || r, s = H.from(n.headers);
  let i = n.data;
  return l.forEach(e, function(c) {
    i = c.call(r, i, s.normalize(), t ? t.status : void 0);
  }), s.normalize(), i;
}
function Gt(e) {
  return !!(e && e.__CANCEL__);
}
function ue(e, t, r) {
  _.call(this, e ?? "canceled", _.ERR_CANCELED, t, r), this.name = "CanceledError";
}
l.inherits(ue, _, {
  __CANCEL__: !0
});
function In(e, t, r) {
  const n = r.config.validateStatus;
  !r.status || !n || n(r.status) ? e(r) : t(new _(
    "Request failed with status code " + r.status,
    [_.ERR_BAD_REQUEST, _.ERR_BAD_RESPONSE][Math.floor(r.status / 100) - 4],
    r.config,
    r.request,
    r
  ));
}
const Mn = L.isStandardBrowserEnv ? (
  // Standard browser envs support document.cookie
  function() {
    return {
      write: function(r, n, s, i, a, c) {
        const p = [];
        p.push(r + "=" + encodeURIComponent(n)), l.isNumber(s) && p.push("expires=" + new Date(s).toGMTString()), l.isString(i) && p.push("path=" + i), l.isString(a) && p.push("domain=" + a), c === !0 && p.push("secure"), document.cookie = p.join("; ");
      },
      read: function(r) {
        const n = document.cookie.match(new RegExp("(^|;\\s*)(" + r + ")=([^;]*)"));
        return n ? decodeURIComponent(n[3]) : null;
      },
      remove: function(r) {
        this.write(r, "", Date.now() - 864e5);
      }
    };
  }()
) : (
  // Non standard browser env (web workers, react-native) lack needed support.
  function() {
    return {
      write: function() {
      },
      read: function() {
        return null;
      },
      remove: function() {
      }
    };
  }()
);
function zn(e) {
  return /^([a-z][a-z\d+\-.]*:)?\/\//i.test(e);
}
function Hn(e, t) {
  return t ? e.replace(/\/+$/, "") + "/" + t.replace(/^\/+/, "") : e;
}
function Xt(e, t) {
  return e && !zn(t) ? Hn(e, t) : t;
}
const $n = L.isStandardBrowserEnv ? (
  // Standard browser envs have full support of the APIs needed to test
  // whether the request URL is of the same origin as current location.
  function() {
    const t = /(msie|trident)/i.test(navigator.userAgent), r = document.createElement("a");
    let n;
    function s(i) {
      let a = i;
      return t && (r.setAttribute("href", a), a = r.href), r.setAttribute("href", a), {
        href: r.href,
        protocol: r.protocol ? r.protocol.replace(/:$/, "") : "",
        host: r.host,
        search: r.search ? r.search.replace(/^\?/, "") : "",
        hash: r.hash ? r.hash.replace(/^#/, "") : "",
        hostname: r.hostname,
        port: r.port,
        pathname: r.pathname.charAt(0) === "/" ? r.pathname : "/" + r.pathname
      };
    }
    return n = s(window.location.href), function(a) {
      const c = l.isString(a) ? s(a) : a;
      return c.protocol === n.protocol && c.host === n.host;
    };
  }()
) : (
  // Non standard browser envs (web workers, react-native) lack needed support.
  function() {
    return function() {
      return !0;
    };
  }()
);
function qn(e) {
  const t = /^([-+\w]{1,25})(:?\/\/|:)/.exec(e);
  return t && t[1] || "";
}
function Wn(e, t) {
  e = e || 10;
  const r = new Array(e), n = new Array(e);
  let s = 0, i = 0, a;
  return t = t !== void 0 ? t : 1e3, function(p) {
    const u = Date.now(), d = n[i];
    a || (a = u), r[s] = p, n[s] = u;
    let m = i, R = 0;
    for (; m !== s; )
      R += r[m++], m = m % e;
    if (s = (s + 1) % e, s === i && (i = (i + 1) % e), u - a < t)
      return;
    const g = d && u - d;
    return g ? Math.round(R * 1e3 / g) : void 0;
  };
}
function Ct(e, t) {
  let r = 0;
  const n = Wn(50, 250);
  return (s) => {
    const i = s.loaded, a = s.lengthComputable ? s.total : void 0, c = i - r, p = n(c), u = i <= a;
    r = i;
    const d = {
      loaded: i,
      total: a,
      progress: a ? i / a : void 0,
      bytes: c,
      rate: p || void 0,
      estimated: p && a && u ? (a - i) / p : void 0,
      event: s
    };
    d[t ? "download" : "upload"] = !0, e(d);
  };
}
const Vn = typeof XMLHttpRequest < "u", Jn = Vn && function(e) {
  return new Promise(function(r, n) {
    let s = e.data;
    const i = H.from(e.headers).normalize(), a = e.responseType;
    let c;
    function p() {
      e.cancelToken && e.cancelToken.unsubscribe(c), e.signal && e.signal.removeEventListener("abort", c);
    }
    l.isFormData(s) && (L.isStandardBrowserEnv || L.isStandardBrowserWebWorkerEnv ? i.setContentType(!1) : i.setContentType("multipart/form-data;", !1));
    let u = new XMLHttpRequest();
    if (e.auth) {
      const g = e.auth.username || "", y = e.auth.password ? unescape(encodeURIComponent(e.auth.password)) : "";
      i.set("Authorization", "Basic " + btoa(g + ":" + y));
    }
    const d = Xt(e.baseURL, e.url);
    u.open(e.method.toUpperCase(), Jt(d, e.params, e.paramsSerializer), !0), u.timeout = e.timeout;
    function m() {
      if (!u)
        return;
      const g = H.from(
        "getAllResponseHeaders" in u && u.getAllResponseHeaders()
      ), b = {
        data: !a || a === "text" || a === "json" ? u.responseText : u.response,
        status: u.status,
        statusText: u.statusText,
        headers: g,
        config: e,
        request: u
      };
      In(function(C) {
        r(C), p();
      }, function(C) {
        n(C), p();
      }, b), u = null;
    }
    if ("onloadend" in u ? u.onloadend = m : u.onreadystatechange = function() {
      !u || u.readyState !== 4 || u.status === 0 && !(u.responseURL && u.responseURL.indexOf("file:") === 0) || setTimeout(m);
    }, u.onabort = function() {
      u && (n(new _("Request aborted", _.ECONNABORTED, e, u)), u = null);
    }, u.onerror = function() {
      n(new _("Network Error", _.ERR_NETWORK, e, u)), u = null;
    }, u.ontimeout = function() {
      let y = e.timeout ? "timeout of " + e.timeout + "ms exceeded" : "timeout exceeded";
      const b = e.transitional || Yt;
      e.timeoutErrorMessage && (y = e.timeoutErrorMessage), n(new _(
        y,
        b.clarifyTimeoutError ? _.ETIMEDOUT : _.ECONNABORTED,
        e,
        u
      )), u = null;
    }, L.isStandardBrowserEnv) {
      const g = (e.withCredentials || $n(d)) && e.xsrfCookieName && Mn.read(e.xsrfCookieName);
      g && i.set(e.xsrfHeaderName, g);
    }
    s === void 0 && i.setContentType(null), "setRequestHeader" in u && l.forEach(i.toJSON(), function(y, b) {
      u.setRequestHeader(b, y);
    }), l.isUndefined(e.withCredentials) || (u.withCredentials = !!e.withCredentials), a && a !== "json" && (u.responseType = e.responseType), typeof e.onDownloadProgress == "function" && u.addEventListener("progress", Ct(e.onDownloadProgress, !0)), typeof e.onUploadProgress == "function" && u.upload && u.upload.addEventListener("progress", Ct(e.onUploadProgress)), (e.cancelToken || e.signal) && (c = (g) => {
      u && (n(!g || g.type ? new ue(null, e, u) : g), u.abort(), u = null);
    }, e.cancelToken && e.cancelToken.subscribe(c), e.signal && (e.signal.aborted ? c() : e.signal.addEventListener("abort", c)));
    const R = qn(d);
    if (R && L.protocols.indexOf(R) === -1) {
      n(new _("Unsupported protocol " + R + ":", _.ERR_BAD_REQUEST, e));
      return;
    }
    u.send(s || null);
  });
}, Ee = {
  http: bn,
  xhr: Jn
};
l.forEach(Ee, (e, t) => {
  if (e) {
    try {
      Object.defineProperty(e, "name", { value: t });
    } catch {
    }
    Object.defineProperty(e, "adapterName", { value: t });
  }
});
const Yn = {
  getAdapter: (e) => {
    e = l.isArray(e) ? e : [e];
    const { length: t } = e;
    let r, n;
    for (let s = 0; s < t && (r = e[s], !(n = l.isString(r) ? Ee[r.toLowerCase()] : r)); s++)
      ;
    if (!n)
      throw n === !1 ? new _(
        `Adapter ${r} is not supported by the environment`,
        "ERR_NOT_SUPPORT"
      ) : new Error(
        l.hasOwnProp(Ee, r) ? `Adapter '${r}' is not available in the build` : `Unknown adapter '${r}'`
      );
    if (!l.isFunction(n))
      throw new TypeError("adapter is not a function");
    return n;
  },
  adapters: Ee
};
function Ue(e) {
  if (e.cancelToken && e.cancelToken.throwIfRequested(), e.signal && e.signal.aborted)
    throw new ue(null, e);
}
function Pt(e) {
  return Ue(e), e.headers = H.from(e.headers), e.data = Le.call(
    e,
    e.transformRequest
  ), ["post", "put", "patch"].indexOf(e.method) !== -1 && e.headers.setContentType("application/x-www-form-urlencoded", !1), Yn.getAdapter(e.adapter || Ve.adapter)(e).then(function(n) {
    return Ue(e), n.data = Le.call(
      e,
      e.transformResponse,
      n
    ), n.headers = H.from(n.headers), n;
  }, function(n) {
    return Gt(n) || (Ue(e), n && n.response && (n.response.data = Le.call(
      e,
      e.transformResponse,
      n.response
    ), n.response.headers = H.from(n.response.headers))), Promise.reject(n);
  });
}
const At = (e) => e instanceof H ? e.toJSON() : e;
function Z(e, t) {
  t = t || {};
  const r = {};
  function n(u, d, m) {
    return l.isPlainObject(u) && l.isPlainObject(d) ? l.merge.call({ caseless: m }, u, d) : l.isPlainObject(d) ? l.merge({}, d) : l.isArray(d) ? d.slice() : d;
  }
  function s(u, d, m) {
    if (l.isUndefined(d)) {
      if (!l.isUndefined(u))
        return n(void 0, u, m);
    } else
      return n(u, d, m);
  }
  function i(u, d) {
    if (!l.isUndefined(d))
      return n(void 0, d);
  }
  function a(u, d) {
    if (l.isUndefined(d)) {
      if (!l.isUndefined(u))
        return n(void 0, u);
    } else
      return n(void 0, d);
  }
  function c(u, d, m) {
    if (m in t)
      return n(u, d);
    if (m in e)
      return n(void 0, u);
  }
  const p = {
    url: i,
    method: i,
    data: i,
    baseURL: a,
    transformRequest: a,
    transformResponse: a,
    paramsSerializer: a,
    timeout: a,
    timeoutMessage: a,
    withCredentials: a,
    adapter: a,
    responseType: a,
    xsrfCookieName: a,
    xsrfHeaderName: a,
    onUploadProgress: a,
    onDownloadProgress: a,
    decompress: a,
    maxContentLength: a,
    maxBodyLength: a,
    beforeRedirect: a,
    transport: a,
    httpAgent: a,
    httpsAgent: a,
    cancelToken: a,
    socketPath: a,
    responseEncoding: a,
    validateStatus: c,
    headers: (u, d) => s(At(u), At(d), !0)
  };
  return l.forEach(Object.keys(Object.assign({}, e, t)), function(d) {
    const m = p[d] || s, R = m(e[d], t[d], d);
    l.isUndefined(R) && m !== c || (r[d] = R);
  }), r;
}
const Zt = "1.4.0", Je = {};
["object", "boolean", "number", "function", "string", "symbol"].forEach((e, t) => {
  Je[e] = function(n) {
    return typeof n === e || "a" + (t < 1 ? "n " : " ") + e;
  };
});
const jt = {};
Je.transitional = function(t, r, n) {
  function s(i, a) {
    return "[Axios v" + Zt + "] Transitional option '" + i + "'" + a + (n ? ". " + n : "");
  }
  return (i, a, c) => {
    if (t === !1)
      throw new _(
        s(a, " has been removed" + (r ? " in " + r : "")),
        _.ERR_DEPRECATED
      );
    return r && !jt[a] && (jt[a] = !0, console.warn(
      s(
        a,
        " has been deprecated since v" + r + " and will be removed in the near future"
      )
    )), t ? t(i, a, c) : !0;
  };
};
function Kn(e, t, r) {
  if (typeof e != "object")
    throw new _("options must be an object", _.ERR_BAD_OPTION_VALUE);
  const n = Object.keys(e);
  let s = n.length;
  for (; s-- > 0; ) {
    const i = n[s], a = t[i];
    if (a) {
      const c = e[i], p = c === void 0 || a(c, i, e);
      if (p !== !0)
        throw new _("option " + i + " must be " + p, _.ERR_BAD_OPTION_VALUE);
      continue;
    }
    if (r !== !0)
      throw new _("Unknown option " + i, _.ERR_BAD_OPTION);
  }
}
const He = {
  assertOptions: Kn,
  validators: Je
}, q = He.validators;
class ge {
  constructor(t) {
    this.defaults = t, this.interceptors = {
      request: new xt(),
      response: new xt()
    };
  }
  /**
   * Dispatch a request
   *
   * @param {String|Object} configOrUrl The config specific for this request (merged with this.defaults)
   * @param {?Object} config
   *
   * @returns {Promise} The Promise to be fulfilled
   */
  request(t, r) {
    typeof t == "string" ? (r = r || {}, r.url = t) : r = t || {}, r = Z(this.defaults, r);
    const { transitional: n, paramsSerializer: s, headers: i } = r;
    n !== void 0 && He.assertOptions(n, {
      silentJSONParsing: q.transitional(q.boolean),
      forcedJSONParsing: q.transitional(q.boolean),
      clarifyTimeoutError: q.transitional(q.boolean)
    }, !1), s != null && (l.isFunction(s) ? r.paramsSerializer = {
      serialize: s
    } : He.assertOptions(s, {
      encode: q.function,
      serialize: q.function
    }, !0)), r.method = (r.method || this.defaults.method || "get").toLowerCase();
    let a;
    a = i && l.merge(
      i.common,
      i[r.method]
    ), a && l.forEach(
      ["delete", "get", "head", "post", "put", "patch", "common"],
      (y) => {
        delete i[y];
      }
    ), r.headers = H.concat(a, i);
    const c = [];
    let p = !0;
    this.interceptors.request.forEach(function(b) {
      typeof b.runWhen == "function" && b.runWhen(r) === !1 || (p = p && b.synchronous, c.unshift(b.fulfilled, b.rejected));
    });
    const u = [];
    this.interceptors.response.forEach(function(b) {
      u.push(b.fulfilled, b.rejected);
    });
    let d, m = 0, R;
    if (!p) {
      const y = [Pt.bind(this), void 0];
      for (y.unshift.apply(y, c), y.push.apply(y, u), R = y.length, d = Promise.resolve(r); m < R; )
        d = d.then(y[m++], y[m++]);
      return d;
    }
    R = c.length;
    let g = r;
    for (m = 0; m < R; ) {
      const y = c[m++], b = c[m++];
      try {
        g = y(g);
      } catch (D) {
        b.call(this, D);
        break;
      }
    }
    try {
      d = Pt.call(this, g);
    } catch (y) {
      return Promise.reject(y);
    }
    for (m = 0, R = u.length; m < R; )
      d = d.then(u[m++], u[m++]);
    return d;
  }
  getUri(t) {
    t = Z(this.defaults, t);
    const r = Xt(t.baseURL, t.url);
    return Jt(r, t.params, t.paramsSerializer);
  }
}
l.forEach(["delete", "get", "head", "options"], function(t) {
  ge.prototype[t] = function(r, n) {
    return this.request(Z(n || {}, {
      method: t,
      url: r,
      data: (n || {}).data
    }));
  };
});
l.forEach(["post", "put", "patch"], function(t) {
  function r(n) {
    return function(i, a, c) {
      return this.request(Z(c || {}, {
        method: t,
        headers: n ? {
          "Content-Type": "multipart/form-data"
        } : {},
        url: i,
        data: a
      }));
    };
  }
  ge.prototype[t] = r(), ge.prototype[t + "Form"] = r(!0);
});
const be = ge;
class Ye {
  constructor(t) {
    if (typeof t != "function")
      throw new TypeError("executor must be a function.");
    let r;
    this.promise = new Promise(function(i) {
      r = i;
    });
    const n = this;
    this.promise.then((s) => {
      if (!n._listeners)
        return;
      let i = n._listeners.length;
      for (; i-- > 0; )
        n._listeners[i](s);
      n._listeners = null;
    }), this.promise.then = (s) => {
      let i;
      const a = new Promise((c) => {
        n.subscribe(c), i = c;
      }).then(s);
      return a.cancel = function() {
        n.unsubscribe(i);
      }, a;
    }, t(function(i, a, c) {
      n.reason || (n.reason = new ue(i, a, c), r(n.reason));
    });
  }
  /**
   * Throws a `CanceledError` if cancellation has been requested.
   */
  throwIfRequested() {
    if (this.reason)
      throw this.reason;
  }
  /**
   * Subscribe to the cancel signal
   */
  subscribe(t) {
    if (this.reason) {
      t(this.reason);
      return;
    }
    this._listeners ? this._listeners.push(t) : this._listeners = [t];
  }
  /**
   * Unsubscribe from the cancel signal
   */
  unsubscribe(t) {
    if (!this._listeners)
      return;
    const r = this._listeners.indexOf(t);
    r !== -1 && this._listeners.splice(r, 1);
  }
  /**
   * Returns an object that contains a new `CancelToken` and a function that, when called,
   * cancels the `CancelToken`.
   */
  static source() {
    let t;
    return {
      token: new Ye(function(s) {
        t = s;
      }),
      cancel: t
    };
  }
}
const Gn = Ye;
function Xn(e) {
  return function(r) {
    return e.apply(null, r);
  };
}
function Zn(e) {
  return l.isObject(e) && e.isAxiosError === !0;
}
const $e = {
  Continue: 100,
  SwitchingProtocols: 101,
  Processing: 102,
  EarlyHints: 103,
  Ok: 200,
  Created: 201,
  Accepted: 202,
  NonAuthoritativeInformation: 203,
  NoContent: 204,
  ResetContent: 205,
  PartialContent: 206,
  MultiStatus: 207,
  AlreadyReported: 208,
  ImUsed: 226,
  MultipleChoices: 300,
  MovedPermanently: 301,
  Found: 302,
  SeeOther: 303,
  NotModified: 304,
  UseProxy: 305,
  Unused: 306,
  TemporaryRedirect: 307,
  PermanentRedirect: 308,
  BadRequest: 400,
  Unauthorized: 401,
  PaymentRequired: 402,
  Forbidden: 403,
  NotFound: 404,
  MethodNotAllowed: 405,
  NotAcceptable: 406,
  ProxyAuthenticationRequired: 407,
  RequestTimeout: 408,
  Conflict: 409,
  Gone: 410,
  LengthRequired: 411,
  PreconditionFailed: 412,
  PayloadTooLarge: 413,
  UriTooLong: 414,
  UnsupportedMediaType: 415,
  RangeNotSatisfiable: 416,
  ExpectationFailed: 417,
  ImATeapot: 418,
  MisdirectedRequest: 421,
  UnprocessableEntity: 422,
  Locked: 423,
  FailedDependency: 424,
  TooEarly: 425,
  UpgradeRequired: 426,
  PreconditionRequired: 428,
  TooManyRequests: 429,
  RequestHeaderFieldsTooLarge: 431,
  UnavailableForLegalReasons: 451,
  InternalServerError: 500,
  NotImplemented: 501,
  BadGateway: 502,
  ServiceUnavailable: 503,
  GatewayTimeout: 504,
  HttpVersionNotSupported: 505,
  VariantAlsoNegotiates: 506,
  InsufficientStorage: 507,
  LoopDetected: 508,
  NotExtended: 510,
  NetworkAuthenticationRequired: 511
};
Object.entries($e).forEach(([e, t]) => {
  $e[t] = e;
});
const Qn = $e;
function Qt(e) {
  const t = new be(e), r = kt(be.prototype.request, t);
  return l.extend(r, be.prototype, t, { allOwnKeys: !0 }), l.extend(r, t, null, { allOwnKeys: !0 }), r.create = function(s) {
    return Qt(Z(e, s));
  }, r;
}
const j = Qt(Ve);
j.Axios = be;
j.CanceledError = ue;
j.CancelToken = Gn;
j.isCancel = Gt;
j.VERSION = Zt;
j.toFormData = _e;
j.AxiosError = _;
j.Cancel = j.CanceledError;
j.all = function(t) {
  return Promise.all(t);
};
j.spread = Xn;
j.isAxiosError = Zn;
j.mergeConfig = Z;
j.AxiosHeaders = H;
j.formToJSON = (e) => Kt(l.isHTMLForm(e) ? new FormData(e) : e);
j.HttpStatusCode = Qn;
j.default = j;
const eo = j;
function to(e) {
  return ae({ tag: "svg", attr: { viewBox: "0 0 24 24", fill: "none" }, child: [{ tag: "path", attr: { opacity: "0.2", fillRule: "evenodd", clipRule: "evenodd", d: "M12 19C15.866 19 19 15.866 19 12C19 8.13401 15.866 5 12 5C8.13401 5 5 8.13401 5 12C5 15.866 8.13401 19 12 19ZM12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22Z", fill: "currentColor" } }, { tag: "path", attr: { d: "M2 12C2 6.47715 6.47715 2 12 2V5C8.13401 5 5 8.13401 5 12H2Z", fill: "currentColor" } }] })(e);
}
function ro(e) {
  return ae({ tag: "svg", attr: { fill: "currentColor", viewBox: "0 0 16 16" }, child: [{ tag: "path", attr: { fillRule: "evenodd", d: "M3.112 5.112a3.125 3.125 0 0 0-.17.613C1.266 6.095 0 7.555 0 9.318 0 11.366 1.708 13 3.781 13H11L3.112 5.112zm11.372 7.372L4.937 2.937A5.512 5.512 0 0 1 8 2c2.69 0 4.923 2 5.166 4.579C14.758 6.804 16 8.137 16 9.773a3.2 3.2 0 0 1-1.516 2.711zm-.838 1.87-12-12 .708-.708 12 12-.707.707z" } }] })(e);
}
function no(e) {
  return ae({ tag: "svg", attr: { fill: "currentColor", viewBox: "0 0 16 16" }, child: [{ tag: "path", attr: { d: "M8 2a5.53 5.53 0 0 0-3.594 1.342c-.766.66-1.321 1.52-1.464 2.383C1.266 6.095 0 7.555 0 9.318 0 11.366 1.708 13 3.781 13h8.906C14.502 13 16 11.57 16 9.773c0-1.636-1.242-2.969-2.834-3.194C12.923 3.999 10.69 2 8 2zm2.354 4.854-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 1 1 .708-.708L7 8.793l2.646-2.647a.5.5 0 0 1 .708.708z" } }] })(e);
}
const so = ({
  children: e,
  apiKey: t,
  filename: r,
  maxFileSize: n,
  allowedFileTypes: s,
  hideAttribution: i,
  onUploadComplete: a,
  accentColor: c = "#f35815"
}) => {
  const [p, u] = X(!1), d = Tr(null), [m, R] = X(null), [g, y] = X(!1), [b, D] = X(!1), [C, T] = X(!1), $ = async () => {
    const N = new FormData();
    N.append("file", m), r && N.append("filename", r), n && N.append("maxFileSize", n), s && N.append("allowedFileTypes", s.join(","));
    try {
      y(!0);
      const { data: V } = await eo.post(
        "https://api.uploadfly.cloud/upload",
        N,
        {
          headers: {
            Authorization: `Bearer ${t}`
          }
        }
      );
      D(!0), a && a(V);
    } catch (V) {
      T(!0), console.log(V);
    } finally {
      y(!1);
    }
  }, Y = () => {
    g || (u(!1), R(null), T(!1), D(!1));
  }, [xe, ee] = X(!1);
  return /* @__PURE__ */ O.jsxs(
    "div",
    {
      className: "uf_widget_container",
      style: { "--accent": c },
      onDragOver: (N) => {
        N.preventDefault(), ee(!0);
      },
      onDragLeave: (N) => {
        N.preventDefault(), ee(!1);
      },
      onDrop: (N) => {
        N.preventDefault(), ee(!1);
        const V = N.dataTransfer.files[0];
        R(V);
      },
      children: [
        /* @__PURE__ */ O.jsx("div", { onClick: () => u(!0), children: e }),
        p && /* @__PURE__ */ O.jsx(
          "div",
          {
            className: "uf_widget_modal",
            style: {
              pointerEvents: p ? "auto" : "none"
            },
            children: /* @__PURE__ */ O.jsx("div", { className: "uf_modal_backdrop rounded-md", onClick: Y, children: /* @__PURE__ */ O.jsxs(
              "div",
              {
                className: `uf_modal_container ${xe ? "drag-over" : ""}`,
                onClick: (N) => N.stopPropagation(),
                children: [
                  /* @__PURE__ */ O.jsx("button", { onClick: Y, className: "uf_close_button", children: /* @__PURE__ */ O.jsx(Dr, {}) }),
                  /* @__PURE__ */ O.jsxs(
                    "div",
                    {
                      style: {
                        width: "100%"
                      },
                      children: [
                        /* @__PURE__ */ O.jsx(
                          "input",
                          {
                            type: "file",
                            ref: d,
                            hidden: !0,
                            multiple: !1,
                            onChange: (N) => R(N.target.files[0])
                          }
                        ),
                        !m && !C && !b && /* @__PURE__ */ O.jsxs("div", { className: "uf_choose_a_file", children: [
                          /* @__PURE__ */ O.jsx(
                            "button",
                            {
                              className: "uf_choose_a_file_btn",
                              onClick: () => d.current.click(),
                              children: "Choose a file"
                            }
                          ),
                          /* @__PURE__ */ O.jsx("p", { className: "uf_choose_a_file_text", children: "...or drop a file" })
                        ] }),
                        m && !C && !b && /* @__PURE__ */ O.jsxs("div", { className: "uf_file_selected", children: [
                          !g && /* @__PURE__ */ O.jsx(
                            "button",
                            {
                              className: "uf_change_file",
                              onClick: () => d.current.click(),
                              children: "Change file"
                            }
                          ),
                          /* @__PURE__ */ O.jsxs("div", { className: "uf_file_info", children: [
                            /* @__PURE__ */ O.jsxs("p", { children: [
                              /* @__PURE__ */ O.jsx("span", { children: "Name: " }),
                              m.name
                            ] }),
                            /* @__PURE__ */ O.jsxs("p", { children: [
                              /* @__PURE__ */ O.jsx("span", { children: "Type: " }),
                              m.type
                            ] }),
                            /* @__PURE__ */ O.jsxs("p", { children: [
                              /* @__PURE__ */ O.jsx("span", { children: "Size: " }),
                              Br(m.size).human("si")
                            ] })
                          ] }),
                          /* @__PURE__ */ O.jsxs(
                            "button",
                            {
                              className: "uf_upload_button",
                              disabled: g,
                              onClick: $,
                              children: [
                                g ? /* @__PURE__ */ O.jsx(to, { className: "uf_animate_spin" }) : /* @__PURE__ */ O.jsx(Fr, {}),
                                "Upload",
                                g && "ing..."
                              ]
                            }
                          )
                        ] }),
                        C && /* @__PURE__ */ O.jsxs("div", { className: "uf_file_upload_response", children: [
                          /* @__PURE__ */ O.jsx(
                            ro,
                            {
                              style: {
                                color: "rgb(220, 38, 38)",
                                fontSize: "4.5rem",
                                lineHeight: 1
                              }
                            }
                          ),
                          /* @__PURE__ */ O.jsx("h2", { className: "", children: "File upload failed" })
                        ] }),
                        b && /* @__PURE__ */ O.jsxs("div", { className: "uf_file_upload_response", children: [
                          /* @__PURE__ */ O.jsx(
                            no,
                            {
                              style: {
                                color: "rgb(22, 163, 74)",
                                fontSize: "4.5rem",
                                lineHeight: 1
                              }
                            }
                          ),
                          /* @__PURE__ */ O.jsx("h2", { children: "File upload successful" })
                        ] })
                      ]
                    }
                  ),
                  !i && /* @__PURE__ */ O.jsxs("p", { className: "uf_attribution", children: [
                    "Powered with  by",
                    " ",
                    /* @__PURE__ */ O.jsx(
                      "a",
                      {
                        href: "https://uploadfly.cloud?utm_source=react_widget",
                        target: "_blank",
                        children: "Uploadfly"
                      }
                    )
                  ] })
                ]
              }
            ) })
          }
        )
      ]
    }
  );
};
export {
  so as UfWidget
};
